src/ikpy/link.py

```diff
@@ -30,7 +30,13 @@ def _get_rotation_axis(self):
         # Defaults to None
         return [0, 0, 0, 1]
 
-    def get_transformation_matrix(self, theta):
+    def get_transformation_params(self):
+        return 0
+
+    def get_bounds(self):
+        return []
+    
+    def get_transformation_matrix(self):
```

Comment:
Actually, this is one of the big change I made to the Link and Chain classes. Now `get_transfomration_matrix` can have 0, 1 or several parameters (as said in the previous PR, which I closed). I think changing the Link interface is not harmful for backward compatibility. But now you can't count the number of parameters of a Chain by counting the number of links. So, I added a method to Chain to get the number of parameters (`get_num_params`).
The interface is a little changed, but may be for a better understanding for the user ?